BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -560,50 +673,97 @@ extension StackLayout {
-        case .unconstrained:
-            maxCross = crossMagnitudes.reduce(0, max)
-        case .exactly(let exactConstraint):
-            maxCross = exactConstraint
-        case .atMost(let maxConstraint):
-            let maxMagnitude = crossMagnitudes.reduce(0, max)
-            maxCross = min(maxConstraint, maxMagnitude)
-        }
-
-        // Finally, form segments from the magnitudes and the alignment option
-        let segments = zip(items, crossMagnitudes).map { (item, measuredCross) -> Segment in
-            let origin: CGFloat
-            let magnitude: CGFloat
-
-            switch alignment {
-            case .center:
-                origin = (maxCross - measuredCross) / 2.0
-                magnitude = measuredCross
-
-            case .fill:
-                origin = 0.0
-                magnitude = maxCross
-
-            case .leading:
-                origin = 0.0
-                magnitude = measuredCross
-
-            case .trailing:
-                origin = maxCross - measuredCross
-                magnitude = measuredCross
+        func fillSegments() -> [Segment] {
+            // Determine the available cross based on the constraint
+            let availableCross: CGFloat
+            switch crossConstraint {
+            case .unconstrained:
+                availableCross = crossMagnitudes.reduce(0, max)
+            case .exactly(let exactConstraint):
+                availableCross = exactConstraint
+            case .atMost(let maxConstraint):
+                let maxMagnitude = crossMagnitudes.reduce(0, max)
+                availableCross = min(maxConstraint, maxMagnitude)
             }
 
-            return Segment(origin: origin, magnitude: magnitude)
+            // Form segments that fill the available space
+            let segments = Array(
+                repeating: Segment(origin: 0, magnitude: availableCross),
+                count: items.count
+            )
+
+            return segments
+        }
+
+        func alignSegments(to alignment: AlignmentID.Type) -> [Segment] {
+            // Get the alignment values for each child
+            let alignmentValues = items.indices.map { i -> CGFloat in
+                let measuredCross = crossMagnitudes[i]
+                let axisSize = axisConstraints[i]
+
+                let size = Vector(axis: axisSize, cross: measuredCross).size(axis: axis)
+                let dimensions = ElementDimensions(size: size)
+                let alignmentGuide = items[i].traits.alignmentGuide
+
+                let value = alignmentGuide?.computeValue(dimensions) ?? alignment.defaultValue(in: dimensions)
+
+                return value
+            }
+
+            // Find the relative distance from the "lowest" edge to the "highest" edge.
+            // This may be greater than the size of any single child, if alignment values push
+            // children outside of their normal bounds.
+
+            let minAlignedCross = -alignmentValues.max()!
```

Comment:
Any reason to be concerned about performance in these max calls for many elements?

Reply:
(Eg, is this called once per layout, or N times, one for each item?)

Reply:
Shouldn't be a problem. `max` is O(n), so it's the same iterating over the children again, which we do several times during a stack layout. `alignSegments` is only called once per layout.